src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -7091,6 +7091,7 @@ internal static bool PathIsNetworkPath(string path)
             /// <param name="symbolicLinkFlags">Flag values from SymbolicLinkFlags enum.</param>
             /// <returns>1 on successful creation.</returns>
             [DllImport(PinvokeDllNames.CreateSymbolicLinkDllName, CharSet = CharSet.Unicode, SetLastError = true)]
+            [return: MarshalAs(UnmanagedType.I1)]
```

Comment:
I wonder why we move from return int to get the marshaling?

Reply:
@iSazonov Because int was incorrect. It implies a four byte return, but the signature is 
```C
BOOLEAN CreateSymbolicLinkW(
  LPCWSTR lpSymlinkFileName,
  LPCWSTR lpTargetFileName,
  DWORD   dwFlags
);
```
BOOLEAN is a typedef for BYTE, which is a single byte. marshalling-wise, it is a byte, but logically a bool. And the way to express that in C# native interop, is by using a bool and the 
```CSharp 
[return: MarshalAs(UnmanagedType.I1)]
```
attribute.
This is the kind of bugs that happen to work on certain platforms, and then manifest in weird ways. I think the condition was incorrect to, `result == 1`, since the documentation says 
> If the function succeeds, the return value is nonzero.

But that will work fine with a `bool`. Any non-zero will be true.

Reply:
I see. Many thanks for fixing this!
Could you contribute more often? :-)